weblate/api/tests.py

```diff
@@ -1217,6 +1218,295 @@ def test_create_component_local(self):
+            code=201,
+            superuser=True,
+            request={
+                "name": "API project",
+                "slug": "api-project",
+                "web": "https://weblate.org/",
+            },
+        ).data
+        self.client.post(
+            reverse(
+                "api:glossary-projects",
+                kwargs={"id": Glossary.objects.order_by("id").all()[0].id},
+            ),
+            {"project_slug": project["slug"]},
+        )
+        response = self.client.delete(
+            reverse(
+                "api:glossary-delete-projects",
+                kwargs={
+                    "id": Glossary.objects.order_by("id").all()[0].id,
+                    "project_slug": -1,
+                },
+            ),
+        )
+        self.assertEqual(response.status_code, 400)
+        response = self.client.delete(
+            reverse(
+                "api:glossary-delete-projects",
+                kwargs={
+                    "id": Glossary.objects.order_by("id").all()[0].id,
+                    "project_slug": project["slug"],
+                },
+            ),
+        )
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(len(Glossary.objects.order_by("id").all()[0].links.all()), 0)
+
+    def test_post_terms(self):
+        self.do_request(
+            "api:glossary-terms",
+            kwargs={"id": Glossary.objects.order_by("id").all()[0].id},
+            method="post",
+            superuser=True,
+            code=201,
+            format="json",
+            request={"language": {"code": "cs"}, "source": "test", "target": "test"},
+        )
+        self.assertEqual(Glossary.objects.order_by("id").all()[0].term_set.count(), 1)
+
+    def test_list_terms(self):
+        self.do_request(
+            "api:glossary-terms",
+            kwargs={"id": Glossary.objects.order_by("id").all()[0].id},
+            method="post",
+            superuser=True,
+            code=201,
+            format="json",
+            request={"language": {"code": "cs"}, "source": "test", "target": "test"},
+        )
+        response = self.do_request(
+            "api:glossary-terms",
+            kwargs={"id": Glossary.objects.order_by("id").all()[0].id},
+            method="get",
+            superuser=True,
+            code=200,
+        )
+        self.assertEqual(response.data["count"], 1)
+
+    def test_get_terms(self):
+        term = self.do_request(
+            "api:glossary-terms",
+            kwargs={"id": Glossary.objects.order_by("id").all()[0].id},
+            method="post",
+            superuser=True,
+            code=201,
+            format="json",
+            request={"language": {"code": "cs"}, "source": "test", "target": "test"},
+        ).data
+        self.do_request(
+            "api:glossary-terms-details",
+            kwargs={"id": Glossary.objects.order_by("id").all()[0].id, "term_id": -1},
+            method="get",
+            code=400,
```

Comment:
Non existing term should fail with 404

```suggestion
            code=404,
```